src/app/data-planes/data/index.ts

```diff
@@ -44,12 +59,28 @@ export type DataplaneOverview = PartialDataplaneOverview & {
 
 const DataplaneNetworking = {
   fromObject(partialDataplaneNetworking: PartialDataplaneNetworking): DataplaneNetworking {
-    // TODO: Normalize inbound and outbound omitempty arrays.
-    // TODO: Consider to determine inbound address and service address here.
-    // Address: `${inbound.address ?? props.data.dataplane.networking.advertisedAddress ?? props.data.dataplane.networking.address}:${inbound.port}`
-    // Service address: `${inbound.serviceAddress ?? inbound.address ?? props.data.dataplane.networking.address}:${inbound.servicePort ?? inbound.port}`
+    const { inbound, outbound, ...rest } = partialDataplaneNetworking
+
+    const inbounds: DataplaneInbound[] = (inbound ?? []).map((inbound) => {
+      // An inbound without a health property is considered healthy.
+      const health = { ready: !inbound.health || inbound.health.ready }
+      const addressPort = `${inbound.address ?? partialDataplaneNetworking.advertisedAddress ?? partialDataplaneNetworking.address}:${inbound.port}`
+      const serviceAddressPort = `${inbound.serviceAddress ?? inbound.address ?? partialDataplaneNetworking.address}:${inbound.servicePort ?? inbound.port}`
+
```

Comment:
For the properties we use to create our new properties, do we want to make sure that the possibly strings are always strings (i.e. `''` if unset)?

Reply:
You mean `inbound.address`, `partialDataplaneNetworking.advertisedAddress`, etc? What would be the  benefit of that?

Reply:
Well at a minimum its always a `string` not a string some of the time. Which is half the point of this line of work.

I think with these types of properties whether to use `string` or `string | undefined` is case by case but I think it makes sense to use `string` for this one. Don't worry though, feel free to resolve if not.

Reply:
Arguably `string | undefined` is the safer type because it gives you one more surface to consider the question: Can I just use this?

Sure, with `string | undefined`, a lot of things will be the same as if you’d just use `string` with the "empty" value being the empty string. For example, nothing in Vue’s templating will complain to render out `{{ inbound.address }}` and end up with the string "undefined" on the page. However, in some cases, you do get extra safety through type checks: passing a value of type `string | undefined` to a function accepting only `string` will error. Switching to using the empty string for such values is therefore worse because now it’s only your memory making sure (or not) that you checks first if you can use the value.

I’m going to resolve this.